web_src/js/modules/fomantic/dropdown.js

```diff
@@ -94,6 +94,22 @@ function delegateOne($dropdown) {
     updateSelectionLabel($label[0]);
     return $label;
   });
+
+  const oldSet = dropdownCall('internal', 'set');
+  const oldSetDirection = oldSet.direction;
+  oldSet.direction = function($menu) {
+    oldSetDirection.call(this, $menu);
+    const classNames = dropdownCall('setting', 'className');
+    $menu = $menu || $dropdown.find('> .menu');
+    const elMenu = $menu[0];
+    // detect whether the menu is outside the viewport, and adjust the position
+    // there is a bug in fomantic's builtin `direction` function, in some cases (when the menu width is only a little larger) it wrongly opens the menu at right and triggers the scrollbar.
+    elMenu.classList.add(classNames.loading);
```

Comment:
What's the purpose of this loading class? Surely the code below is fast enough to not need it?

Reply:
Ok, it does not show a spinner, right?

Reply:
Actually I think it couldn't as we have remove the `loading` module already 😆

Reply:
That's totally different. You could read related CSS styles and `direction` code, then these questions could be answered.

Reply:
> Ok, it does not show a spinner, right?

No. It is only used for calculating the size.

Reply:
Ok. And no, I'm not too thrilled to read Fomantic code right now 😆.